pkg/route/registry/route/rest.go

```diff
@@ -71,6 +74,15 @@ func (rs *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, er
 		return nil, errors.NewConflict("route", route.Namespace, fmt.Errorf("Route.Namespace does not match the provided context"))
 	}
 
+	shard, allocError := rs.allocator.Allocate(route)
```

Comment:
It's ok to redefine err.  In fact, it's encouraged (don't let it carry over deeper).  The go pattern is very much "deal with errors right away by returning", at which point err is free.  Go code should encourage "err" to always be the most recent error in the function except in wierd cases.


> On Mar 12, 2015, at 1:47 PM, ramr <notifications@github.com> wrote:
> 
> In pkg/route/registry/route/rest.go:
> 
> > @@ -71,6 +74,15 @@ func (rs *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, er
> >  		return nil, errors.NewConflict("route", route.Namespace, fmt.Errorf("Route.Namespace does not match the provided context"))
> >  	}
> >  
> > +	shard, allocError := rs.allocator.Allocate(route)
> Yeah, the err and errs names were already used up in that function.
> 
> —
> Reply to this email directly or view it on GitHub.
>